ui/src/AudioRecorder.js

```diff
@@ -1,44 +1,48 @@
 export class AudioRecorder {
     /**
-    * Creates an AudioRecorder that can record utterances
-    * @param {String} stores and allows access to utterance after recording
-    * @param {Object} allows access to Web Audio API
-    */
+     * Creates an AudioRecorder that can record utterances
+     * @param {String} recordingUrl - A URL that allows access to an audio file after recording.
+     * @param {Blob} blob - An audio recording file. 
+     * @param {Object} mediaRecorder - Allows access to Web Audio API.
+     */
     constructor() {
         this.recordingUrl;
+        this.blob;
         this.mediaRecorder;
     }
+
     /**
-    * Prompts user for access to Microphone using API and starts recording is access granted
-    */
+     * Prompts user for access to Microphone Component and begins recording if access is granted.
+     */
     startRecording() {
         navigator.mediaDevices.getUserMedia({ audio: true, video: false })
             .then((stream) => {
                 const mediaRecorder = new MediaRecorder(stream);
                 this.mediaRecorder = mediaRecorder;
                 mediaRecorder.start();
             }).catch(function(err) {
-                console.log(err.name, err.message); // Is this good to have or is just the alert fine?
-                alert("need access to microphone to record");
+                alert("Error: Need access to microphone to record.");
```

Comment:
The error is thrown if they don't accept the permissions? Is there anyway to recover from this? Can user try again?

Reply:
After much Google research, the answer unfortunately seems to be no. This seems to be a very common issue with no known workaround--hitting "Block" in Chrome adds the site to a blocked list, and there does not appear to be a manual way for us to remove it from the blocked list. If you remove it from the blocked list, it will reprompt the next time the user presses the record button. People online suggested including a how-to-do-this after the error message (here's the most helpful thread I found regarding the issue: https://stackoverflow.com/questions/15993581/reprompt-for-permissions-with-getusermedia-after-initial-denial), so I will add this in the error message for now